include/revng/Model/CommonTypeMethods.h

```diff
@@ -101,28 +132,41 @@ class CommonTypeMethods {
   const model::PointerType *tryGetAsPointer() const;
   bool isPointer() const { return tryGetAsPointer() != nullptr; }
 
+  /// This helper returns true if and only if this type is is a primitive or
+  /// a pointer.
+  ///
+  /// \note it asserts on `void`.
   bool isScalar() const;
 
 public:
   /// Use this method to check whether a specific type is constant or not, as
   /// opposed to the \ref IsConst() member (which should be reserved for
   /// assigning `const`ness), since it unwraps typedefs and arrays.
-  ///
-  /// \note If you ever need a way to get rid of constness, please introduce
-  ///       a dedicated `dropConst` helper here as opposed to doing that at
-  ///       the call site.
   bool isConst() const;
 
   /// Skip all the "wrappers" around the definition.
   ///
+  /// This is provided for some very specific cases, please take extra care when
+  /// using it. If you are hesitating whether you need this, please shop around
+  /// other functions provided here. If any of them sounds like it should do
+  /// the trick, prefer it instead.
```

Comment:
1-2. 👌 
3. Because sadness that's why. In practically every case we want `skipToDefinition` except for one: the place where we want to manually set the root for the reference deserialized from an IR string. It would be too sad to make every spot use inferior version just because of a one ~~dodgy~~ bad apple. I guess we can technically move it closer to its only user, but what if somebody at some point also needs it? Having it here decreases the chance of a copy getting reimplemented.